src/abstract_text_transformer.jl

```diff
@@ -0,0 +1,38 @@
+abstract type AbstractTextTransformer <: MMI.Unsupervised end
+
+## General method to fit text transformer models ##
+MMI.fit(transformer::AbstractTextTransformer, verbosity::Int, X) = 
+    _fit(transformer, verbosity, build_corpus(X))
+
+function _fit(transformer::AbstractTextTransformer, verbosity::Int, X::Corpus)
+    transformer.max_doc_freq < transformer.min_doc_freq && 
```

Comment:
Testing validity of hyper-parameters should go into a `clean!` method. This method is called automatically at construction (because you are using `@mlj_model` macro define your structs, right?) but is also called by MLJ's `fit!` method every time it is called - so there is no need for your `MMI.fit` to check them again. 

It looks like you haven't got `clean!` methods yet. Below is an example from the [docs](https://alan-turing-institute.github.io/MLJ.jl/dev/adding_models_for_general_use/#New-model-type-declarations-and-optional-clean!-method)

```julia
function MMI.clean!(model::RidgeRegressor)
    warning = ""
    if model.lambda < 0
        warning *= "Need lambda ≥ 0. Resetting lambda=0. "
        model.lambda = 0
    end
    return warning
end
```

Reply:
Thanks for the suggestion - I went with option 2.  Testing it seems to work ok.